TypeCobol/Compiler/CodeModel/SymbolTable.cs

```diff
@@ -824,6 +805,26 @@ public void AddType(TypeDefinition type)
             Add(Types, type);
         }
 
+        public void AddTypeDataDefinition(TypeDefinition data)
+        {
+            //Add type if it doesn't have children
+            if (data.Children.Count == 0)
+            {
+                Add(DataTypeEntries, data);
+            }
+
+            //If type has children, add only the children
+            foreach (var elem in data.Children)
```

Comment:
Why not also copy children of children recursively ?

Reply:
They will be copied, as the method is called at each level of a TypeDef

Reply:
It's not the case.
All children of children of an intrinsic type are not recognized anymore.

Reply:
OK, I will add recursively the children's children